### DISCUSSION TITLE
Sorting dates written as strings

### DISCUSSION THREAD

**Discussion by Tex-Tang** [observablehq/plot#1328] [Q&A] [answered]
I want to make a monthly bar chart. My data structure as below
```
[ { month: "Jan 21", value: 100 }, ...]
```
```javascript
{
  marks: [
    Plot.barY(sampleData, {
      x: "month",
      y: "value",
      sort: null,
    }),
    Plot.ruleY([0]),
  ],
  x: {
    label: "Month",
  },
}
```

The final result as shown below, the months are sorted alphabetically. I tried multiple ways to customize the sort function but it doesn't work.

![image](https://user-images.githubusercontent.com/14901260/224536312-4a277f70-23c2-4105-b5a1-a860a2b5d5a7.png)

**Comment by Fil** [comment, marked as answer]
Two suggestions: the first is to read the dates first as JavaScript dates, for example with d3.utcParse, and use an interval to recreate the bars as rect marks. The second is to keep them as strings (and ordinal values, represented by bars), but transform them so that they are ordered naturally by time.
See the two possibilities at https://observablehq.com/@recifs/reading-dates
